src/examples/treemap/treemap.ts

```diff
@@ -0,0 +1,203 @@
+  const arr: any[] = []
+  for (const k in obj) {
+    if (k === '__data') {
+      continue
+    }
+    const child: any = {
+      name: k,
+      depth,
+      children: descend(obj[k], depth + 1)
+    }
+    if ('__data' in obj[k]) {
+      child.data = obj[k].__data
+    }
+    arr.push(child)
+  }
+  return arr
+}
+
+function burrow(table: Row[]) {
+  // create nested object
+  const obj: any = {}
+
+  table.forEach((row: Row) => {
+    // start at root
+    let layer = obj
+
+    // create children as nested objects
+    row.taxonomy.value.forEach((key: any) => {
+      layer[key] = key in layer ? layer[key] : {}
+      layer = layer[key]
+    })
+    layer.__data = row
+  })
+
+  // use descend to create nested children arrays
+  return {
+    name: 'root',
+    children: descend(obj, 1),
+    depth: 0
+  }
+}
+
+const vis: TreemapVisualization = {
+  id: 'treemap',
+  label: 'Treemap',
+  options: {
+    color_range: {
+      type: 'array',
+      label: 'Color Range',
+      display: 'colors',
+      default: ['#dd3333', '#80ce5d', '#f78131', '#369dc1', '#c572d3', '#36c1b3', '#b57052', '#ed69af']
+    }
+  },
+  // Set up the initial state of the visualization
+  create: function(element, config) {
+    this.svg = d3.select(element).append('svg')
+  },
+  // Render in response to the data or settings changing
+  update: function(data, element, config, queryResponse) {
+    if (!handleErrors(this, queryResponse, {
+      min_pivots: 0, max_pivots: 0,
+      min_dimensions: 1, max_dimensions: undefined,
+      min_measures: 1, max_measures: 1
+    })) return
+
+    const width = element.clientWidth
+    const height = element.clientHeight
+
+    const dimensions = queryResponse.fields.dimension_like
+    const measure = queryResponse.fields.measure_like[0]
+
+    const format = formatType(measure.value_format) || ((s: any): string => s.toString())
+
+    const color = d3.scaleOrdinal().range(config.color_range)
+    // const color = d3.scaleOrdinal().range(config.color_range || this.options.color_range.default) // DNR
+
+    data.forEach(function(row) {
+      row.taxonomy = {
+        value: dimensions.map((dimension) => row[dimension.name].value)
+      }
+    })
+
+    const treemap = d3.treemap()
+        .size([width, height - 16])
+        .tile(d3.treemapSquarify.ratio(1))
+        .paddingOuter(1)
+				.paddingTop((d) => {
+          return d.depth === 1 ? 16 : 0
+        })
+        .paddingInner(1)
+        .round(true)
+
+    const svg = this.svg!
+      .html('')
+      .attr('width', '100%')
+      .attr('height', '100%')
+      .append('g')
+      .attr('transform', 'translate(0,16)')
+
+    const breadcrumb = svg.append('text')
+      .attr('y', -5)
+      .attr('x', 4)
+
+    const root = d3.hierarchy(burrow(data)).sum((d: any) => {
+      return 'data' in d ? d.data[measure.name].value : 0
+    })
+    treemap(root)
+
+    const cell = svg.selectAll('.node')
+        .data(root.descendants())
+      .enter().append('g')
```

Comment:
Needs an indent

Reply:
Rather, a lot of this file is indented inconsistently compared to the rest of the project. Looks like some lines use tabs?

Reply:
yeah, i think this was pre-existing in the JS version. It's following some d3 indentation conventions where certain chained methods (.enter, .exit) are outdented relative to the methods that follow them. I'm not really a fan of these conventions, myself, but I don't know if it's our place to make stylistic recommendations. Maybe we should get some guidance on this from looker. @nickfogler what do you think?

Reply:
Pretty sure their style is a 2-space indent, and there's a `tslint` config in this repo that they defined. I'd go with whatever makes `tslint` happy.